File: How_To_Program_Exercises/JavaHowToProgram9/Chapter4/CreditLimits.java
/********************************************************
*	Credit Limits										*
*	Program that will determine whether a department-	*
*	store customer has exceeded the credit limit on a	*
*	charge account										*
*********************************************************/

import java.util.Scanner;

class CreditLimits
{
	public static void main (String[] args)
	{
		Scanner input = new Scanner (System.in);
		int accountNumber, balance, totalCharges, totalCredits, creditLimit;
		
		System.out.print("\n\nEnter account number (or -1 to quit):  ");
		accountNumber = input.nextInt();
		
		while ( accountNumber != -1 )
		{
			System.out.print("Enter the beginning balance: ");
			balance = input.nextInt();
			
			System.out.print("Enter total charges: ");
			totalCharges = input.nextInt();
			
			System.out.print("Enter total credits: ");
			totalCredits = input.nextInt();
			
			System.out.print("Enter credit limit: ");
			creditLimit = input.nextInt();
			
			balance += totalCharges - totalCredits;
			
			System.out.printf("New balance is %d\n", balance);

			if (balance > creditLimit)
			{
				System.out.printf ("Account: %d\n", accountNumber);
				System.out.printf ("Credit limit: %d\n", creditLimit);
				System.out.printf ("Balance: %d\n", balance);
				System.out.printf ("Credit Limit Exceeded.\n\n");
			}
			
			System.out.print("\n\nEnter account number (or -1 to quit):  ");
			accountNumber = input.nextInt();
		}
	}
}

File: How_To_Program_Exercises/Visual_C#_How_To_Program2012_Fifth/Chapter5/CreditLimits.cs
/********************************************************
*	Credit Limits										*
*	Program that will determine whether a department-	*
*	store customer has exceeded the credit limit on a	*
*	charge account										*
*********************************************************/

using System;

class CreditLimits
{
	public static void Main (string[] args)
	{
		int accountNumber, balance, totalCharges, totalCredits, creditLimit;
		
		Console.Write("\n\nEnter account number (or -1 to quit):  ");
		accountNumber = Convert.ToInt32(Console.ReadLine());
		
		while ( accountNumber != -1 )
		{
			Console.Write("Enter the beginning balance: ");
			balance = Convert.ToInt32(Console.ReadLine());
			
			Console.Write("Enter total charges: ");
			totalCharges = Convert.ToInt32(Console.ReadLine());
			
			Console.Write("Enter total credits: ");
			totalCredits = Convert.ToInt32(Console.ReadLine());
			
			Console.Write("Enter credit limit: ");
			creditLimit = Convert.ToInt32(Console.ReadLine());
			
			balance += totalCharges - totalCredits;
			
			Console.WriteLine("New balance is {0}", balance);
			
			if (balance > creditLimit)
			{
				Console.WriteLine ("Account: {0}", accountNumber);
				Console.WriteLine ("Credit limit: {0}", creditLimit);
				Console.WriteLine ("Balance: {0}", balance);
				Console.WriteLine ("Credit Limit Exceeded.\n\n");
			}
			
			Console.Write("\n\nEnter account number (or -1 to quit):  ");
			accountNumber = Convert.ToInt32(Console.ReadLine());
		}
	}
}